server/src/test/java/org/elasticsearch/indices/IndicesRequestCacheTests.java

```diff
@@ -441,19 +532,51 @@ public void testEqualsKey() throws IOException {
         IndexReader reader2 = DirectoryReader.open(writer);
         IndexReader.CacheKey rKey2 = reader2.getReaderCacheHelper().getKey();
         IOUtils.close(reader1, reader2, writer, dir);
-        IndicesRequestCache.Key key1 = new IndicesRequestCache.Key(new TestEntity(null, trueBoolean), rKey1, new TestBytesReference(1));
-        IndicesRequestCache.Key key2 = new IndicesRequestCache.Key(new TestEntity(null, trueBoolean), rKey1, new TestBytesReference(1));
-        IndicesRequestCache.Key key3 = new IndicesRequestCache.Key(new TestEntity(null, falseBoolean), rKey1, new TestBytesReference(1));
-        IndicesRequestCache.Key key4 = new IndicesRequestCache.Key(new TestEntity(null, trueBoolean), rKey2, new TestBytesReference(1));
-        IndicesRequestCache.Key key5 = new IndicesRequestCache.Key(new TestEntity(null, trueBoolean), rKey1, new TestBytesReference(2));
-        String s = "Some other random object";
-        assertEquals(key1, key1);
+        List<IndicesRequestCache.Key> keys = new ArrayList<>();
+        for (AtomicBoolean bool : new AtomicBoolean[] { trueBoolean, falseBoolean }) {
+            for (MappingLookup.CacheKey mKey : new MappingLookup.CacheKey[] { mKey1, mKey2 }) {
+                for (IndexReader.CacheKey rKey : new IndexReader.CacheKey[] { rKey1, rKey2 }) {
+                    for (BytesReference requestKey : new BytesReference[] { new TestBytesReference(1), new TestBytesReference(2) }) {
+                        keys.add(new IndicesRequestCache.Key(new TestEntity(null, bool), mKey, rKey, requestKey));
+                    }
+                }
+            }
+        }
+        for (IndicesRequestCache.Key key : keys) {
+            assertNotEquals(key, null);
+            assertNotEquals(key, "Some other random object");
+        }
+        for (IndicesRequestCache.Key key1 : keys) {
+            assertNotEquals(key1, null);
+            for (IndicesRequestCache.Key key2 : keys) {
+                if (key1 == key2) {
+                    assertEquals(key1, key2);
+                    assertEquals(key1.hashCode(), key2.hashCode());
```

Comment:
we have EqualsHashCodeTestUtils, not sure it's worth using it here

Reply:
Me neither given that we already had all this stuff to test it.